sqld/src/http/mod.rs

```diff
@@ -272,25 +270,68 @@ pub async fn run_http<D: Database>(
                 .allow_methods(cors::AllowMethods::any())
                 .allow_headers(cors::Any)
                 .allow_origin(cors::Any),
-        )
-        .service_fn(move |req| {
-            handle_request(
-                auth.clone(),
-                req,
-                upgrade_tx.clone(),
-                hrana_http_srv.clone(),
-                db_factory.clone(),
-                enable_console,
-                stats.clone(),
-            )
-        });
+        );
 
     let listener = tokio::net::TcpListener::bind(&addr).await?;
-    let server = hyper::server::Server::builder(AddrIncoming::from_listener(listener)?)
+    hyper::server::Server::builder(AddrIncoming::from_listener(listener)?)
         .tcp_nodelay(true)
-        .serve(tower::make::Shared::new(service));
-
-    server.await.context("Http server exited with an error")?;
+        .serve(layered_app.into_make_service())
+        .await
+        .context("foo")?;
 
     Ok(())
 }
+
+/// Axum authenticated extractor
+#[tonic::async_trait]
```

Comment:
@LucioFranco just curious: technically `axum` could also deliver a sync variant of those traits, in case somebody doesn't need async, right? Since I see that our implementations of the required traits are in fact all 100% await-free.

Reply:
Axum is mean't to be used 100% in an async context. Mostly this is because hyper's api that axum uses even though you can use something else than hyper.

Reply:
Makes sense, I just wondered if it wouldn't be a performance win to allow a regular sync trait here, given that async trait has this hidden allocation cost, wrapping the future in a Box